resolwe_bio/processes/clustering/pca.yml

```diff
@@ -45,38 +45,27 @@
     runtime: polyglot
     language: bash
     program: |
-      {% for e in exps %}
-        {% if e.exp_type != exps.0.exp_type %}
-          re-error "All expressions must be of the same expression type."
-        {% endif %}
-        {% if genes %}
-          {%if e.source != genes_source %}
-            re-error "Source of selected genes must be the same as the source of all expression files."
-          {% endif %}
-        {% else %}
-          {% if e.source != exps.0.source %}
-            re-error "All expressions must have the same source."
-          {% endif %}
-        {% endif %}
-      {% endfor %}
-
-      sample_files=()
-      sample_ids=()
+      {% if exps|selectattr('source', 'equalto', (exps|first).source)|list|length != exps|length %}
```

Comment:
Why did you refactor this?

Reply:
No particular reason. Can be changed back if `for` loops are preferred over one-liners.